TESTS/netsocket/tcp/tcpsocket_echotest.cpp

```diff
@@ -96,32 +96,28 @@ void TCPSOCKET_ECHOTEST()
 
 void tcpsocket_echotest_nonblock_receive()
 {
-    while (bytes2recv > 0) {
-        int recvd = sock.recv(&(tcp_global::rx_buffer[bytes2recv_total - bytes2recv]), bytes2recv);
-        if (recvd == NSAPI_ERROR_WOULD_BLOCK) {
-            if (tc_exec_time.read() >= time_allotted) {
-                TEST_FAIL();
-                receive_error = true;
-                tx_sem.release();
-                return;
-            }
-            continue;
-        } else if (recvd < 0) {
-            TEST_FAIL();
+    int recvd = sock.recv(&(tcp_global::rx_buffer[bytes2recv_total - bytes2recv]), bytes2recv);
```

Comment:
I don't think you should remove the while() loop.

You are supposed to loop UNTIl you get the WOULD_BLOCK.

Only then you are guaranteed to receive sigio() event.

Reply:
Not sure I understand correctly... The `tcpsocket_echotest_nonblock_receive()` function is called from sigio handler. I thought it is supposed to trigger on the event to be sure the data is there waiting to be read?
Perhaps I am missing something about how the sigio works?

Reply:
For example, if the driver received 2kB of data.. sends `sigio()` event to notify that there is new data to be read, and you read 1kB. It does not send another event, because there was not change in the state.. the device is still readable.

Reply:
`sigio()` notifies that there is change of state.

readable -> not readable or
not readable -> readable or
connected -> disconnected or
connecting -> connected or
new connections acceptable -> no new connections
etc...

Reply:
Now I see. Then I will also fix the tlssocket echo tests as I made the same mistake there some time ago.

Reply:
Brough the loop back into the receiver both for tcp and tls socket tests. Thanks for pointing this out!